network-config-analyzer/SchemeRunner.py

```diff
@@ -137,13 +142,16 @@ def run_scheme(self):
         :return: The number of queries with unexpected result + number of configs with unexpected number of warnings
         :rtype: int
         """
-        allowed_keys = {'networkConfigList': 1, 'namespaceList': 0, 'podList': 0, 'queries': 0}
+        allowed_keys = {'networkConfigList': 1, 'namespaceList': 0, 'podList': 0, 'serviceList': 0, 'queries': 0}
         self.check_fields_validity(self.scheme, 'scheme', allowed_keys)
 
         # global resource files
         pod_list = self._handle_resources_list(self.scheme.get('podList', 'k8s'))
         ns_list = self._handle_resources_list(self.scheme.get('namespaceList', 'k8s'))
         peer_container = PeerContainer(ns_list, pod_list)
+        # look for service resources under 'pod_list' files
+        services = K8sServiceYamlParser.parse_service_resources(pod_list, peer_container)
```

Comment:
Why 2 calls for `parse_service_resources` from scheme runner?

Reply:
As far as I understand from the code, ns resources and pod resources may be global (per scheme), or per network config. Since we read service resources from pod list, we must try to read them per config as well, when supplied.